src/session.jl

```diff
@@ -6,26 +6,35 @@ module session
 export GoogleSession, authorize
 
 import Base: string, print, show
-using Dates, Base64 
+using Dates, Base64
 
 #using Requests
-using HTTP, HTTP.Messages 
+using HTTP, HTTP.Messages
 
-import JSON
-import MbedTLS
+using JSON: JSON
+using MbedTLS: MbedTLS
 
 using ..error
 using ..credentials
 using ..root
 
+global mbedtlslock = ReentrantLock()
 """
     SHA256withRSA(message, key)
 
 Sign message using private key with RSASSA-PKCS1-V1_5-SIGN algorithm.
 """
-SHA256withRSA(message, key::MbedTLS.PKContext) = MbedTLS.sign(key, MbedTLS.MD_SHA256,
-    MbedTLS.digest(MbedTLS.MD_SHA256, message), MbedTLS.MersenneTwister(0)
-)
+function SHA256withRSA(message, key::MbedTLS.PKContext)
+    lock(mbedtlslock)
+    output = MbedTLS.sign(
+        key,
+        MbedTLS.MD_SHA256,
+        MbedTLS.digest(MbedTLS.MD_SHA256, message),
+        MbedTLS.MersenneTwister(0),
+    )
+    unlock(mbedtlslock)
+    return output
```

Comment:
```suggestion
    @lock MBEDTLSLOCK begin
        output = MbedTLS.sign(
            key,
            MbedTLS.MD_SHA256,
            MbedTLS.digest(MbedTLS.MD_SHA256, message),
            MbedTLS.MersenneTwister(0),
        )
    end
    return output
```
using `@lock` here ensures the `begin-end` block is wrapped in a `try-finally`. This is important because if the `MbedTLS.sign` call were to throw an error for some reason, our global lock here would be "stuck" and no further signing could happen.